src/cdomains/intDomain.ml

```diff
@@ -3144,7 +3176,19 @@ struct
 
   let bitor ik x y = bit2 Ints_t.bitor ik x y
 
-  let bitand ik x y = bit2 Ints_t.bitand ik x y
+  let bitand ik x y =  match x, y with
+    | None, None -> None
+    | None, _ | _, None -> raise (ArithmeticOnIntegerBot (Printf.sprintf "%s op %s" (show x) (show y)))
+    | Some (c, m), Some (c', m') ->
+      if (m =: Ints_t.zero && m' =: Ints_t.zero) then
+        (* both arguments constant *)
+        Some (Ints_t.bitand c c', Ints_t.zero)
+      else if m' =: Ints_t.zero && c' =: Ints_t.one && Ints_t.rem m (Ints_t.of_int 2) =: Ints_t.zero then
+        (* x & 1  and  x == c (mod 2*z) *)
+        (* Value is equal to LSB of c *)
+        Some (Ints_t.bitand c c', Ints_t.zero)
```

Comment:
@sim642 I would appreciate it of you can also think this part through again: It seems correct now, but I am not 100% confident.

Reply:
This sounds right, although since `c' = 1`, this really is just `bitand c 1`.

Another way to do this might be to just delegate `x & 1` to `x % 2`. If `m` is even, then this seems to match with `rem` implementation. But `rem` also handles the other case where `m` is odd, so maybe we can just delegate to that?

Reply:
I think that would be incorrect, though?

~~~C

#include<stdlib.h>
#include<stdio.h>

int main(){
  int i  = (-5) %2; // -1
  int j = -5&1; // 1
  printf("%i\n",i);
  printf("%i",j);
  return 0;
}
~~~

Reply:
Turns out we are unsound for `x %2` for signed types when x is maybe negative and not constant. See: https://github.com/goblint/analyzer/issues/1156